frame/staking-rewards/src/lib.rs

```diff
@@ -780,9 +773,98 @@ pub mod pallet {
 			});
 			Ok([*position, new_position])
 		}
+
+		#[transactional]
+		fn claim(who: &Self::AccountId, position: &Self::PositionId) -> DispatchResult {
+			let keep_alive = false;
+
+			let mut stake =
+				Stakes::<T>::try_get(position).map_err(|_| Error::<T>::StakeNotFound)?;
+			ensure!(who == &stake.owner, Error::<T>::OnlyStakeOwnerCanUnstake);
+
+			let pool_id = stake.reward_pool_id;
+			let mut rewards_pool =
+				RewardPools::<T>::try_get(pool_id).map_err(|_| Error::<T>::RewardsPoolNotFound)?;
+
+			(rewards_pool, stake) =
+				Self::collect_rewards(&pool_id, rewards_pool, stake, false, keep_alive)?;
+
+			RewardPools::<T>::insert(pool_id, rewards_pool);
+			Stakes::<T>::insert(position, stake);
```

Comment:
you could consider refactoring this to use `try_mutate` and refactoring `collect_rewards` to take `&mut` instead of passing the value through the function

Reply:
While I agree that will clean up the code. As per the design, `unstake` isn't supposed to update `Stakes`, only remove the position, while `claim` updates `Stakes`. Given that both extrinsics call into this function (reducing duplication), I leave the final storage edits to the calling functions. 

Let me know if I misunderstood the design or if you have a better idea.

If not, should I include cargo docs to explain that this function does not attempt to update storage?

Reply:
19609362a733d2bff55aa8ec5a6a79e97fab7560